### ISSUE TITLE
No longer compatible with terraform v12.0+

### ISSUE THREAD

**Issue by waeltken** [carlpett/tfz53#12]
```
Error: Invalid resource name

  on dns.tf line 1717, in resource "aws_route53_record" "171-some-domain-name-A":
1717: resource "aws_route53_record" "171-some-domain-name-A" {

A name must start with a letter and may contain only letters, digits,
underscores, and dashes.```

**Comment by carlpett** [comment]
Hi @waeltken,
Thanks for reporting! I'll add a flag for controlling syntax version.

**Comment by waeltken** [comment]
I think it should be sufficient to move the counter away from the beginning of the logical name of each resource.

———————————
Clemens Wältken
clemens@waeltken.de

**Comment by carlpett** [comment]
Hm, I just realized that your problem is not at all what I was looking at. The existing code generates syntax which is not 0.12-compatible, but that is not what your error is about, it is about an illegal identifier. It looks like your zone has a record like `171.example.com`, and I currently more or less just swap dots for dashes to create the resource name.
I did a quick check, and TF 0.11 _also_ does not allow for leading digits in the resource id, so this isn't related to the update, I think.

I'll need to do a separate patch to fix this. Should just sanitize by adding some letter at the beginning, I guess.
